engine/src/main/java/org/destinationsol/menu/background/BackgroundShipManager.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.menu.background;
+
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.Body;
+import com.badlogic.gdx.physics.box2d.BodyDef;
+import com.badlogic.gdx.physics.box2d.World;
+import org.destinationsol.assets.Assets;
+import org.destinationsol.assets.json.Validator;
+import org.destinationsol.common.SolColor;
+import org.destinationsol.game.CollisionMeshLoader;
+import org.destinationsol.game.drawables.DrawableLevel;
+import org.destinationsol.ui.DisplayDimensions;
+import org.destinationsol.ui.UiDrawer;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BackgroundShipManager {
+    private DisplayDimensions displayDimensions;
+    private CollisionMeshLoader shipMeshLoader;
+
+    private List<String> nextShips;
+    private List<BackgroundObject> backgroundShips;
+
+    private World world;
+
+    public BackgroundShipManager(DisplayDimensions displayDimensions, World world) {
+
+        this.displayDimensions = displayDimensions;
+        this.world = world;
+
+        shipMeshLoader = new CollisionMeshLoader();
+
+        nextShips = new ArrayList<String>() {
+            //Add ships in the order they are supposed to be displayed
+            {
+                add("core:imperialTiny");
+                add("core:imperialSmall");
+                add("core:minerSmall");
+                add("core:minerBoss");
+                add("core:desertSmall");
+                add("core:desertOrbiter");
+            }
```

Comment:
This would be much cooler if it instead read from a config file. That way a module can add its own ships if it so desires.

Reply:
Modules can now add their ships to the menu by using a `menuBackgroundShipConfig.json`